File: Backend/src/main/java/com/example/riskfactors/model/DNSHealth.java
package com.example.riskfactors.model;

import lombok.*;
import javax.persistence.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.SequenceGenerator;
import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@Entity
@Data
@AllArgsConstructor
@Builder
public class DNSHealth {

    @Id
    @SequenceGenerator(
            name = "DNSHealthSequence",
            sequenceName = "DNSHealthSequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "DNSHealthSequence"
    )
    private Long dnsHealthId;
    private float riskScore;
    private boolean isOpenDNSDetected = false;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(
            name = "spfFK"
    )
    private SPF spf;

    @ElementCollection
    private List<String> ns;
    @ElementCollection
    private List<String> mx;
    @ElementCollection
    private List<String> txt;
    @ElementCollection
    private List<String> a;
    @ElementCollection
    private List<String> quadA;
    @ElementCollection
    private List<String> aName;
    @ElementCollection
    private List<String> cName;
    @ElementCollection
    private List<String> soa;
    @ElementCollection
    private List<String> srv;
    @ElementCollection
    private List<String> ptr;

    public DNSHealth() {
        spf = new SPF();
        this.ns = new ArrayList<>();
        this.mx = new ArrayList<>();
        this.txt = new ArrayList<>();
        this.a = new ArrayList<>();
        this.quadA = new ArrayList<>();
        this.aName = new ArrayList<>();
        this.cName = new ArrayList<>();
        this.soa = new ArrayList<>();
        this.srv = new ArrayList<>();
        this.ptr = new ArrayList<>();

    }
}

File: Frontend/src/app/business-logic/DNSHealth.ts
import { SPF } from "./SPF";
export class DNSHealth {
    public riskScore:number | undefined;
    public isOpenDNSDetected:boolean = false;

    public spf:SPF | undefined;
    public ns:Array<String> | undefined;
    public mx:Array<String> | undefined;;
    public txt:Array<String> | undefined;;
    public a:Array<String> | undefined;;
    public quadA:Array<String> | undefined;;
    public aName:Array<String> | undefined;;
    public cName:Array<String> | undefined;;
    public soa:Array<String> | undefined;;
    public srv:Array<String> | undefined;;
    public ptr:Array<String> | undefined;;

    public DNSHealth() {
        this.spf = new SPF();
        this.ns = new Array<String>;
        this.mx = new Array<String>;
        this.txt = new Array<String>;
        this.a = new Array<String>;
        this.quadA = new Array<String>;
        this.aName = new Array<String>;
        this.cName = new Array<String>;
        this.soa = new Array<String>;
        this.srv = new Array<String>;
        this.ptr = new Array<String>;

    }


}
